client/src/components/Tasks.jsx

```diff
@@ -1,39 +1,43 @@
-import React, { Component } from "react";
+import React from "react";
 import OneInputForm from "./OneInputForm";
 import Task from "./Task";
+import { useListTasks } from '../hooks/useListTasks'
+import { useTasksAndLists } from "../hooks/useTasksAndLists";
 
 function Tasks(props) {
-
 	const listID = props.match.params.id;
-
-	const tasks = props.lists[listID].tasks.map((taskID, index) => {
-		const task = props.tasks[taskID];
-
-		if (!task) return null;
-
-		const { complete, text } = task;
+	// This is another custom hook, that hides logic related to
+	// a certain list in one place.
+	const {
+		title,
+		listTasks,
+		addTaskToList,
+		removeTaskFromList,
+	} = useListTasks(listID);
+	// For toggling lists, we use the shared hook for now.
+	const { toggleTask } = useTasksAndLists();
+
+	// Added an extra length check to avoid an error
+	const tasks = listTasks.length > 0 && listTasks.map(({ id, complete, text }) => {
 		return (
 			<Task
-				key={index}
+				key={id}
 				complete={complete}
-				onDelete={() => props.onDeleteTask(listID, taskID)}
-				onComplete={() => props.onToggleTask(taskID)}
+				onDelete={() => removeTaskFromList(id)}
+				onComplete={() => toggleTask(id)}
 				taskText={text}
 			/>
 		);
 	});
 
 	return (
 		<div className="tasks-page">
-			<h1>{props.lists[listID].title}</h1>
+			<h1>{title}</h1>
```

Comment:
I tried to move logic like this out of the component, and into the hook.
Same thing, different place.